app/gui/src/controller/graph/widget/response.rs

```diff
@@ -138,9 +120,41 @@ pub enum Display {
 #[derive(Debug, serde::Deserialize)]
 pub(super) struct Choice<'a> {
     /// The value of the choice. Must be a valid Enso expression.
-    pub value: Cow<'a, str>,
+    #[serde(borrow)]
+    pub value:      Cow<'a, str>,
     /// Custom label to display in the dropdown. If not provided, IDE will create a label based on
     /// value.
     #[serde(borrow)]
-    pub label: Option<Cow<'a, str>>,
+    pub label:      Option<Cow<'a, str>>,
+    /// A list of widget overrides for nested arguments within this choice's expression.
+    #[serde(borrow, default)]
+    pub parameters: Vec<Argument<'a>>,
+}
+
+/// Widget data for single function argument. The engine usually serializes it as an tuple array of
+/// argument name and widget definition, but sometimes it is incorrectly represented as a Tuple type
+/// constructor, so for now we have to support deserializing both formats.
+#[derive(Debug, serde::Deserialize)]
+pub(super) struct Argument<'a> {
+    #[serde(alias = "first")]
+    pub name: Cow<'a, str>,
+    #[serde(borrow, alias = "second")]
+    pub data: FallableWidgetDefinition<'a>,
+}
+
+/// A wrapper type that allows deserialization of a widget definitions to partially fail: failure
+/// message of individual widget definition deserialization will be preserved and deserialization
+/// will continue.
+#[derive(Debug)]
+pub(super) struct FallableWidgetDefinition<'a> {
+    pub(super) widget: FallibleResult<Option<WidgetDefinition<'a>>>,
+}
+
+impl<'de: 'a, 'a> serde::Deserialize<'de> for FallableWidgetDefinition<'a> {
```

Comment:
Why this can't be derived?

Reply:
Just to be clear, this code was just moved around within the file. The reason why it's not derived is because its special handling of deserialization is really all this type is made for. This type is allowed to gracefuly fail to deserialize, without causing the whole visualization response to be unreadable. That way we only remove the failing widget, while still keeping other ones functional.